### ISSUE TITLE
SSR middleware TS handling process.env error

### ISSUE THREAD

**Issue by Evertvdw** [quasarframework/quasar#8979]
**Describe the bug**
`process.env` variables in SSR middleware (TS) gets compiled to a variable which cannot be found at runtime. Which results in runtime errors in the SSR server.

```
ReferenceError: development is not defined
    at C:\Users\evert\Programming\Mett4Frontend\.quasar\ssr\compiled-middlewares.js:101:5
```

Which comes from this line in my TS middleware file:
`if (process.env.APP_ENV === "staging" || process.env.APP_ENV === "production") {`

which gets compiled to: 
` if (development === "staging" || development === "production") {`

I am using the QEnv app extension (which I know is deprecated) but I do think this issue is more general.  I tried to setup a similar project from a starter which does what I want to do and I get the same error there (using dotenv-flow). 

**To Reproduce**
Run `npm run mett` in this repository: https://github.com/Evertvdw/quasar-v2-ssr-middleware

**Expected behavior**
I used QEnv to set some environment variables (like API endpoints) based on a certain `process.env.QENV` variable in the npm script to run the `quasar build` or `quasar dev` command. I tried something similar with the `dotenv-flow` package (https://www.npmjs.com/package/dotenv-flow) in the repro repository. 

I just want the same behaviour as with the Qenv app extension, and using those variables also in SSR middleware. Anyone got a clue how I can fix this?

**Platform (please complete the following information):**
Quasar Version: v2.0.0-beta.12
@quasar/app Version: v3.0.0-beta.15
Quasar mode:
  - [ ] SPA
  - [x] SSR
  - [ ] PWA
  - [ ] Electron
  - [ ] Cordova
  - [ ] Capacitor
  - [ ] BEX

Tested on:
  - [ ] SPA
  - [x] SSR
  - [ ] PWA
  - [ ] Electron
  - [ ] Cordova
  - [ ] Capacitor
  - [ ] BEX

**Comment by Evertvdw** [comment]
I just changed my middleware from TS to regular js and I still have the same issue so it's probably a webpack thing.

**Comment by rstoenescu** [comment]
Hi,

Stumbled upon this on the webpack5 upgrade work and fixed it already.
Will be available in "@quasar/app" v3.0.0-beta.16.
